server/log_rpc_server.go

```diff
@@ -431,9 +431,24 @@ func (t *TrillianLogRPCServer) GetLatestSignedLogRoot(ctx context.Context, req *
 		FirstTreeSize:  int64(req.FirstTreeSize),
 		SecondTreeSize: int64(root.TreeSize),
 	}
+	if uint64(req.FirstTreeSize) > root.TreeSize {
+		// no need to get consistency proof in this case
```

Comment:
nit: comment unhelpful; it describes what the code is not doing, rather than what's happening.  Maybe something like:
```golang
// Client has asked for a consistency proof to the current root from a size that is bigger than the tree we know about.
// Return an error that includes the SLR we have available (and no consistency proof).
```